index.js

```diff
@@ -44,16 +44,28 @@ function defineFunction(obj, name, fn) {
 
     metrics.start();
 
-    var result = fn.apply(this, arguments);
+    var result;
+    var error;
 
-    if (typeof result.finally === 'function') {
+    try {
+      result = fn.apply(this, arguments);
+    } catch (e) {
```

Comment:
I didn't realize that a `try..finally` without a `catch` would execute and then allow the error to bubble like normal.